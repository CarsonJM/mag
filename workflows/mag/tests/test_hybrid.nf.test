nextflow_workflow {

    name "Test workflow: MAG"
    script "../main.nf"
    workflow "MAG"
    tag "workflows"
    tag "mag"
    tag "mag_test_hybrid"
    config "../../../conf/mag_test_hybrid.config"

    test("Parameters: long reads included") {

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out,
                ).match()}
            )
        }
    }
}
