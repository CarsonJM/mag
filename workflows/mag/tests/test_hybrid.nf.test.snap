{
    "Parameters: long reads included": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "minigut_run0_fastp_1.fastp.fastq.gz:md5,bac85d89ef44638200c247bc941a66fb",
                            "minigut_run0_fastp_2.fastp.fastq.gz:md5,be6b3eb0d336c333e97d925b0e3a0ff6"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "group-0",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-group-0.contigs.fa:md5,c78fa53feb4b35380998bc0c339cdb0e"
                    ],
                    [
                        {
                            "id": "group-0",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdesHybrid"
                        },
                        "SPAdesHybrid-group-0_scaffolds.fasta:md5,84c68c6ba721ad5858588a1487e05b6f"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    "versions.yml:md5,15111ae1ae8b8b570a2efddd7c747bd0",
                    "versions.yml:md5,1bd6bbec156534880e04e0320dbce063",
                    "versions.yml:md5,5de8983775d3ef64cdbec6f382852619",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,b94aaba1559ac9deaf7339181c398405",
                    "versions.yml:md5,cc143ba1b806a335a9963a434c887eba",
                    "versions.yml:md5,d13508b80e270ac6f1543725b431a682",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0",
                    "versions.yml:md5,e2ed4d9a5b5d63ae1a6eef0418d976f0"
                ],
                "assemblies": [
                    [
                        {
                            "id": "group-0",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-group-0.contigs.fa:md5,c78fa53feb4b35380998bc0c339cdb0e"
                    ],
                    [
                        {
                            "id": "group-0",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdesHybrid"
                        },
                        "SPAdesHybrid-group-0_scaffolds.fasta:md5,84c68c6ba721ad5858588a1487e05b6f"
                    ]
                ],
                "bin_summary": [
                    
                ],
                "genomad": [
                    
                ],
                "metaeuk": [
                    
                ],
                "prodigal": [
                    
                ],
                "prokka": [
                    
                ],
                "refined_bins": [
                    
                ],
                "refined_unbins": [
                    
                ],
                "short_reads": [
                    [
                        {
                            "id": "minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "minigut_run0_fastp_1.fastp.fastq.gz:md5,bac85d89ef44638200c247bc941a66fb",
                            "minigut_run0_fastp_2.fastp.fastq.gz:md5,be6b3eb0d336c333e97d925b0e3a0ff6"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15111ae1ae8b8b570a2efddd7c747bd0",
                    "versions.yml:md5,1bd6bbec156534880e04e0320dbce063",
                    "versions.yml:md5,5de8983775d3ef64cdbec6f382852619",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,b94aaba1559ac9deaf7339181c398405",
                    "versions.yml:md5,cc143ba1b806a335a9963a434c887eba",
                    "versions.yml:md5,d13508b80e270ac6f1543725b431a682",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0",
                    "versions.yml:md5,e2ed4d9a5b5d63ae1a6eef0418d976f0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-08T15:47:29.016078996"
    }
}